Cython/Utility/Coroutine.c

```diff
@@ -1501,13 +1501,29 @@ static PyObject *__Pyx_CoroutineAwait_no_new(PyTypeObject *type, PyObject *args,
 }
 #endif
 
+#if PY_VERSION_HEX < 0x03060000
+// In earlier versions of Python an object with no __dict__ and not __slots__ is assumed
+// to be pickleable by default. Coroutine-wrappers have significant state so shouldn't be.
+// Therefore provide a default implementation
+static PyObject *__Pyx_CoroutineAwait_reduce_ex(__pyx_CoroutineAwaitObject *self, PyObject *arg) {
```

Comment:
It turned out that by making it possible to lookup the class correctly, I inadvertently made it possible to pickle the class (which is correct, although likely to fail on unpickling since `_cython_VERSION_NUMBER` isn't a real module...), which inadvertently made it possible to pickle the instance (which is definitely incorrect). These were only impossible to pickle by luck before